#! /bin/sh /usr/share/dpatch/dpatch-run
## 25-strip-and-rpath.dpatch by Jens Peter Secher <jps@debian.org>
##
## DP: Make it possible to override the stripping of the neko executable prior
## DP: to building rest of compiler tools and use -rpath to force search in
## DP: /usr/lib/neko.

@DPATCH@
diff -urNad neko-1.6.0~/Makefile neko-1.6.0/Makefile
--- neko-1.6.0~/Makefile	2007-07-27 00:16:45.000000000 +0200
+++ neko-1.6.0/Makefile	2007-08-18 22:45:53.776335664 +0200
@@ -7,9 +7,10 @@
 MAKESO = $(CC) -shared -WBsymbolic
 LIBNEKO_NAME = libneko.so
 LIBNEKO_LIBS = -ldl -lgc -lm
-NEKOVM_FLAGS = -Lbin -lneko
+NEKOVM_FLAGS = -Lbin -lneko -Wl,-rpath -Wl,/usr/lib/neko
 STD_NDLL_FLAGS = ${NEKOVM_FLAGS}
 INSTALL_FLAGS =
+STRIP=strip
 
 NEKO_EXEC = LD_LIBRARY_PATH=../bin:${LD_LIBRARY_PATH} NEKOPATH=../boot:../bin ../bin/neko
 
@@ -121,7 +122,7 @@
 
 bin/neko: $(VM_OBJECTS)
 	${CC} ${CFLAGS} ${EXTFLAGS} -o $@ ${VM_OBJECTS} ${NEKOVM_FLAGS}
-	strip bin/neko
+	$(STRIP) bin/neko
 
 bin/std.ndll: ${STD_OBJECTS}
 	${MAKESO} -o $@ ${STD_OBJECTS} ${STD_NDLL_FLAGS}
